data/abilities.js

```diff
@@ -309,8 +310,8 @@ exports.BattleAbilities = {
 		shortDesc: "Prevents other Pokemon from lowering this Pokemon's Defense stat stage.",
 		onBoost: function (boost, target, source, effect) {
 			if (source && target === source) return;
-			if (boost['def'] && boost['def'] < 0) {
-				delete boost['def'];
+			if (boost.def && boost.def < 0) {
+				delete boost.def;
```

Comment:
Are these changes actually necessary?

Reply:
Yeah, without them you get the <code>Object is possibly 'undefined'.</code> error.

Reply:
Hm, can you try again with TypeScript Nightly? It should be able to equivocate the two forms...

Reply:
I'm getting the same error with TypeScript Nightly.

Reply:
:[

I guess this can be here. Remind me to screw around with it myself later.